app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java

```diff
@@ -207,7 +220,28 @@ private void loadFragment(int navItemId) {
                 fragment = EventDashboardFragment.newInstance(eventId);
                 break;
         }
-        fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+
+        if(!isAtHome){
+            fragmentManager.popBackStack("root_fragment",0);
+            isAtHome=true;
+        }
+        if(atHome){
+            root_fragment_id = lastSelectedNavItemId;
+            fragment_tag = "root_fragment";
+            fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).addToBackStack(fragment_tag).commit();
+            atHome=false;
+        }
+        else {
+            isAtHome = false;
+            if((lastSelectedNavItemId==R.id.nav_dashboard)||
+                (lastSelectedNavItemId==R.id.nav_events&&root_fragment_id==lastSelectedNavItemId)){
+                root_fragment_id = lastSelectedNavItemId;
+                fragmentManager.popBackStack("root_fragment",FragmentManager.POP_BACK_STACK_INCLUSIVE);
+                fragment_tag = "root_fragment";
```

Comment:
Why is fragment tag mutating so many times and `"root_fragment"` is written twice in string format when you could just use `fragment_tag`

Reply:
fragment_tag is changing as per the changes in the selection of nav items.